apps/s_client.c

```diff
@@ -3241,7 +3241,7 @@ static void print_stuff(BIO *bio, SSL *s, int full)
         }
 
         BIO_printf(bio, "---\n");
-        peer = SSL_get_peer_certificate(s);
+        peer = SSL_get1_peer_certificate(s);
```

Comment:
```suggestion
        peer = SSL_get0_peer_certificate(s);
```
and then delete the `X509_free(peer)` at the bottom of the function.